desidlas/training/model.py

```diff
@@ -0,0 +1,304 @@
+    fc1_n_neurons = hyperparameters['fc1_n_neurons']
+    fc2_1_n_neurons = hyperparameters['fc2_1_n_neurons']
+    fc2_2_n_neurons = hyperparameters['fc2_2_n_neurons']
+    fc2_3_n_neurons = hyperparameters['fc2_3_n_neurons']
+    conv1_kernel = hyperparameters['conv1_kernel']
+    conv2_kernel = hyperparameters['conv2_kernel']
+    conv3_kernel = hyperparameters['conv3_kernel']
+    conv1_filters = hyperparameters['conv1_filters']
+    conv2_filters = hyperparameters['conv2_filters']
+    conv3_filters = hyperparameters['conv3_filters']
+    conv1_stride = hyperparameters['conv1_stride']
+    conv2_stride = hyperparameters['conv2_stride']
+    conv3_stride = hyperparameters['conv3_stride']
+    pool1_kernel = hyperparameters['pool1_kernel']
+    pool2_kernel = hyperparameters['pool2_kernel']
+    pool3_kernel = hyperparameters['pool3_kernel']
+    pool1_stride = hyperparameters['pool1_stride']
+    pool2_stride = hyperparameters['pool2_stride']
+    pool3_stride = hyperparameters['pool3_stride']
+    pool1_method = 1        # Removed from hyperparameter search because it never chose anything but this method
+    pool2_method = 1
+    pool3_method = 1
+
+    #INPUT_SIZE = 600
+    #matrix_size=4
+     #hyperparameters['INPUT_SIZE'] # the pixel of input image,need a new hyperparameter
+     #matrix_size:smoothed data input, change the input training data to 600*4 scale . without smoothing , the matrix_size is 1
+    tfo = {}    # Tensorflow objects
+
+
+    
+
+    """
+
+    x = tf.Variable(tf.ones(shape=[None, INPUT_SIZE]), name='x')
+    label_classifier = tf.Variable(tf.zeros(shape=[None]), name='label_classifier')
+    label_offset = tf.Variable(tf.zeros(shape=[None]), name='label_offset')
+    label_coldensity = tf.Variable(tf.zeros(shape=[None]), name='label_coldensity')
+    keep_prob = tf.Variable( name='keep_prob')
+    global_step = tf.Variable(0, name='global_step', trainable=False)
+
+    """
+
+    # I converted "tf.placeholder" to "tf.Variable"
+
+    tf.compat.v1.disable_eager_execution()
+
+    x = tf.compat.v1.placeholder(tf.float32, shape=[None,matrix_size, INPUT_SIZE], name='x')
+    y = tf.compat.v1.placeholder(tf.float32, shape=[None, matrix_size], name='x')
+
+    label_classifier = tf.compat.v1.placeholder(tf.float32, shape=[None], name='label_classifier')
+    label_offset = tf.compat.v1.placeholder(tf.float32, shape=[None], name='label_offset')
+    label_coldensity = tf.compat.v1.placeholder(tf.float32, shape=[None], name='label_coldensity')
+    keep_prob = tf.compat.v1.placeholder(tf.float32, name='keep_prob')
+    global_step = tf.Variable(0, name='global_step', trainable=False)
+
+    x_4d = tf.reshape(x, [-1, INPUT_SIZE, 1,matrix_size]) #reshape the data size
+    # First Convolutional Layer
+    # Kernel size (16,1)
+    # Stride (4,1)
+    # number of filters = 4 (features?)
+    # Neuron activation = ReLU (rectified linear unit)
+    W_conv1 = weight_variable([conv1_kernel, 1, 4, conv1_filters])
```

Comment:
Can we make the `4` a parameter?

Reply:
I think 4 may not be a variable parameter. I just make the variable parameter when we do the smoothing as matrix_size.